processor/spanmetricsprocessor/cache/cache_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCache(t *testing.T) {
+	type args struct {
+		size int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "create a new Cache with length 10",
+			args: args{
+				size: 10,
+			},
+			wantErr: false,
+		},
+		{
+			name: "create a new Cache with length -1",
+			args: args{
+				size: -1,
+			},
+			wantErr: true,
+		},
+		{
+			name: "create a new Cache with length 0",
+			args: args{
+				size: 0,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
This should be moved to the top of the outer methods since each run inherits the properties of outer t.

Reply:
I think we have to do that inside the test so that the test case would actually be running in parallel. Otherwise only the parent tests are running in parallel but not the nested table-driven tests?

Reply:
![image](https://user-images.githubusercontent.com/2691836/145747119-4b8d6093-57d4-4579-acb7-9abf568535b3.png)

Reply:
> This should be moved to the top of the outer methods since each run inherits the properties of outer t.

This comment was not correct, my apologies. However, I do have a better understanding on how it glues itself together.

`t.Parallel()` still needs to be called on the outer method (to give us the testing speed up we need) and to ensure each top level test case can be started together (instead of sequentially) and those calls to t.Run() are run in a seperate goroutine but in sequential order.

Having each sub test run in sequential order gives little speed up considering there is little work happening within it.

Reply:
I got what you want to achieve now. However, I'm not sure how much benefit we can get out of it.

Golang run tests from different packages in parallel already. We probably are pretty much already taking the most advantage of the CI/CD's computing resource. IMO if the tests are really IO heavy, it might make sense to run the tests in a single *_test.go file in parallel. However, I think it's not our case here.